### PR TITLE
Add produced wares and workers to productionsites encyclopedia

### LINKED ISSUE #3956: Enhance/Clear buildingshelp for production sites
In my opinion the buildingshelp for production sites misses at least one important thing:

How many wares of which type are needed to produce a ware?

Examples: 
* How many logs do a charcoal kiln need to produce one coal?
* How many coal/ores do a smelting works need to produce one ingot of iron/gold?
* How many logs/iron do a toolsmithy need to produce a specific tool?
* How many logs do a sawmill need to produce one plank?
* ...

On the other site the production sites buildingshelp contain information which are maybe superfluous:
* Building space
* Vision range

### PR DESCRIPTION
**Pull request by hessenfarmer**
Based on the work of @frankystone 
Fixes #3956

    pulled out the calculation of "How many wares of which type are needed to produce a ware?" from ware_help.lua and put it in a new file calculations.lua
    adapted ware_help.lua and building_help.lua to use the pulled out function
    fixed documentation for both files
    added calculations.lua to documentation
    fixed the documentation of consumed_wares_workers() and added an example

### CODE REVIEW

File: data/tribes/scripting/help/calculations.lua [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+   -- Find out which programs in the building produce this worker
+   for j, program_name in ipairs(building.production_programs) do
+      for worker, amount in pairs(building:recruited_workers(program_name)) do
+         if (worker_description.name == worker) then
+            table.insert(producing_programs, program_name)
+         end
+      end
+   end
+
+   -- Now collect all wares produced by the filtered programs
```
Comment (tothxa) [inline data/tribes/scripting/help/calculations.lua]:
```suggestion
   -- Now collect all workers recruited by the filtered programs
```

File: data/tribes/scripting/help/calculations.lua [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+--    print("Building: ".. building.name, "Ware: ".. ware_description.name)
+--    for k,v in pairs(deduplicated_programs) do
+--       print("Programs: ", k, v)
+--    end
+--    for k,v in pairs(produced_wares_counters) do
+--       print("wares counter: ", k,v)
+--    end
+--    for k,v in pairs(produced_wares_strings) do
+--       print("ware strings: ", k,v)
+--   end
```
Comment (tothxa) [inline data/tribes/scripting/help/calculations.lua]: Will you still need this debugging block?

File: data/tribes/scripting/help/calculations.lua [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+--    print("Building: ".. building.name, "worker: ".. worker_description.name)
+--    for k,v in pairs(deduplicated_programs) do
+--       print("Programs: ", k, v)
+--    end
+--    for k,v in pairs(recruited_workers_counters) do
+--       print("workers counter: ", k,v)
+--    end
+--    for k,v in pairs(recruited_workers_strings) do
+--       print("worker strings: ", k,v)
+--   end
```
Comment (tothxa) [inline data/tribes/scripting/help/calculations.lua]: Will you still need this debugging block?

**Review by tothxa — COMMENTED** [review]
Thanks to both of you! This is a huge improvement to the encyclopedia.

Tested and works nicely.

**Review by tothxa — APPROVED** [review]

**Comment by frankystone** [comment]
Thanks for implementing this!

**Comment by hessenfarmer** [comment]
@frankystone 
thanks for your work that laid the the basis for this

**Pull request merged**